www/js/lib/uiUtil.js

```diff
@@ -909,11 +929,54 @@ function applyAppTheme (theme) {
         // Note that location.pathname returns the path plus the filename, but is useful because it removes any query string
         var prefix = (window.location.protocol + '//' + window.location.host + window.location.pathname).replace(/\/[^/]*$/, '');
         if (doc) {
+            // Make sure content is hidden while loading if in dark mode
+            htmlEl = document.querySelector('html');
+            var isDark = htmlEl.classList.contains('dark') || 
+                        iframe.classList.contains('_invert') || 
+                        iframe.classList.contains('_wikiVector');
+
+            // Add a blocking style to prevent content from showing before stylesheet loads
+            if (isDark && doc) {
+                // Add inline style to both html and body
+                if (doc.documentElement) {
+                    doc.documentElement.style.backgroundColor = '#e5e5e5';
+                }
+                
+                if (doc.body) {
+                    doc.body.style.backgroundColor = '#e5e5e5';
+                    doc.body.classList.add('content-loading');
+                }
+                
+                // Add a style element with higher priority
+                var blockStyle = doc.createElement('style');
+                blockStyle.textContent = 'html, body { background-color: #e5e5e5 !important; }';
+                blockStyle.id = 'temp-dark-style';
+                doc.head.appendChild(blockStyle);
+            }
```

Comment:
I think all the above code is part of the unsuccessful attempt to avoid Flash of Incorrect Style, right? In which case remove it, and we'll make a new issue to deal with that.